design_docs/Hail_Filter_and_Annotation.md

```diff
@@ -0,0 +1,67 @@
+# Hail Query Stage
+
+This stage of the workflow uses hail query to load, annotate, and apply provisional categories to variants.
+
+[hail_filter_and_classify.py](../reanalysis/hail_filter_and_classify.py)
+
+## Stage Details
+
+Runs within a Hail Batch, as a Hail Query runtime. For now, due to functional constraints, the process runs inside
+DataProc on GCP. We are aiming to split the HailQuery/VEP annotation stage from this infrastructure requirement soon.
+
+This stage takes a Hail MatrixTable representation of a VCF as input, applies VEP annotation, and adds provisional
+category tags. Flow chart representations of the logic for pre-and post annotation filters are shown below
+
+![Pre-Annotation Filters](images/hail_pre_filter.png)
+Annotation with VEP is between these two stages
+
+![Post-Annotation Filters](images/hail_categories.png)
+Criteria for selection into the 4 variant categories
+
+---
+
+### Input
+
+Designed for joint-called MatrixTable. This is due to current reliance on the Seqr-loader
+pipeline at the CPG, which handles the transition from raw reads to clean, joint called, [GATK Best-Practice](https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels-)
+prepared variant data.
+
+If this process is opened up in future so that VCFs or alignment data is used, this will need to be adapted, or a pre-processing step created.
```

Comment:
Yes, would be awesome to have a skippable set of stages here (at some point in the future) that can use raw reads as input.